hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaDeploymentManagement.java

```diff
@@ -689,4 +694,23 @@ public long countActionsAll() {
         return distributionSetRepository.findInstalledAtTarget(controllerId);
     }
 
+    @Override
+    public int deleteActionsByStatusAndLastModifiedBefore(final Set<Status> status, final long lastModified) {
```

Comment:
Be careful this is not a read-only transaction so please use `@Transactional` here

Reply:
Done